galaxykit/groups.py

```diff
@@ -2,38 +2,38 @@
 import json
 
 
-def find_group(galaxy_root, headers, group_name):
+def find_group(client, group_name):
     """
     Returns the data of the group with group_name
     """
-    all_groups = requests.get(f"{galaxy_root}_ui/v1/groups/", headers=headers).json()
-    match_group = [x for x in all_groups["data"] if x["name"] == group_name]
-    return match_group
+    groups_url = f"_ui/v1/groups/?name={group_name}"
+    return client.get(groups_url)
+
+def get_group_id(client, group_name):
+    """
+    Returns the id for a given username
+    """
+    groups_url = f"_ui/v1/groups/?name={group_name}"
+    resp = client.get(groups_url)
+    return resp.json()["data"][0]["id"]
 
 
-def create_group(galaxy_root, headers, group_name):
+def create_group(client, group_name):
     """
     Creates a group
     """
-    group_url = f"{galaxy_root}_ui/v1/groups/"
-    payload = json.dumps({"name": group_name}).encode("utf8")
-    headers = {
-        **headers,
-        "Content-Type": "application/json;charset=utf-8",
-        "Content-length": str(len(payload)),
-    }
-    return requests.post(group_url, headers=headers, data=payload)
+    return client.post("_ui/v1/groups/", {"name": group_name})
```

Comment:
I was initially unsure about the changes to authentication and passing the client around, but seeing how drastically it improves the readability on this function I think I approve.

The complication of seeing `client.post` and not `requests.post` and wondering what's happening behind the scenes is a worthy tradeoff I think for being able to skip repeatedly writing out the headers and payload code snippets.

Reply:
👍